src/nvim/popupmnu.c

```diff
@@ -219,7 +220,9 @@ void pum_display(pumitem_T *array, int size, int selected)
     if (curwin->w_p_rl) {
       pum_width = pum_col - pum_scrollbar + 1;
     } else {
-      pum_width = Columns - pum_col - pum_scrollbar;
+      assert(Columns - pum_col - pum_scrollbar >= INT_MIN
```

Comment:
Okay, thanks for explaining. I won't distract from this issue anymore.

On March 23, 2015 5:04:55 PM EDT, "Eliseo Martínez" <notifications@github.com> wrote:
>> @@ -219,7 +220,9 @@ void pum_display(pumitem_T *array, int size, int
>selected)
>>      if (curwin->w_p_rl) {
>>        pum_width = pum_col - pum_scrollbar + 1;
>>      } else {
>> -      pum_width = Columns - pum_col - pum_scrollbar;
>> +      assert(Columns - pum_col - pum_scrollbar >= INT_MIN
>
>By "shape" I just mean the concrete way of writing a macro, among the
>many possible ones.
>I want the most comfortable one.
>For example, the one you propose is ok, but you have to type `INT_MIN`
>and `INT_MAX` every time.
>It think it's more comfortable something where you just type `INT`, and
>it deduces `INT_MIN` and `INT_MAX` from them. This is:
>```
>#define IN_BOUNDS(type, value) ((value) >= type##_MIN && (value) <=
>type##_MAX)
>```
>It could be named IN_BOUNDS, or IN_RANGE, or whatever else. I don't
>like RANGE alone, cause it doesn't suggest a condition to me, but
>something that gets generated. 
>We could also have an IN_RANGE with generic limits min & max, and
>IN_BOUNDS making use of that for the particular case of checking for
>value in bounds of a given type.
>Possibilities are many. That's why I said it takes a little thinking
>finding the right "shape" for this.
>In any case, we should discuss this in a dedicated issue, better than
>here.
>Let's restrict here to this PR's problems, please.
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/neovim/neovim/pull/2228/files#r26979835